ops/pebble.py

```diff
@@ -1098,11 +1107,16 @@ def __init__(self, name: str, raw: Optional[CheckDict] = None):
         self.name = name
         dct: CheckDict = raw or {}
         self.override: str = dct.get('override', '')
+        self.level: CheckLevel
+        dct_level = dct.get('level', '')
         try:
-            level: Union[CheckLevel, str] = CheckLevel(dct.get('level', ''))
+            self.level = CheckLevel(dct_level)  # CheckLevel('') -> CheckLevel.UNSET
         except ValueError:
-            level = dct.get('level', '')
-        self.level = level
+            warnings.warn(f'Unknown CheckLevel value {dct_level!r}')
+            self.level = CheckLevel.UNKNOWN
+        #
+        # these are all Optional in CheckDict and here, why '' instead of None?
+        # note that all falsey values are filtered out in to_dict
         self.period: Optional[str] = dct.get('period', '')
         self.timeout: Optional[str] = dct.get('timeout', '')
         self.threshold: Optional[int] = dct.get('threshold')
```

Comment:
In general a lot of values are marked as `Optional` when they could perhaps be simply left out of the `TypedDict`. I haven't fully checked what the values received from Pebble for these dicts can be -- can these items be the `go` version of `None` (`null`? `nil`?). It seems like if so, we should use `None` instead of `''`, and if not we shouldn't use `Optional`.